### PR TITLE
doc: matmul: updated supported data types and minor edits

### PR DESCRIPTION
**Pull request by raistefintel**
Matmul supported data types requested by https://github.com/oneapi-src/oneDNN/issues/2196

### CODE REVIEW

**Comment by vpirogov** [comment]
@ranukund, could you please review for style?

File: doc/programming_model/data_types.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ oneDNN supports training and inference with the following data types:
     model implementation.
 
 @note
-    f64 is only supported for convolution, reorder, layer normalization and
+    f64 is only supported for matmul, convolution, reorder, layer normalization and
```
Comment (ranukund) [inline doc/programming_model/data_types.md:46]:
```suggestion
    f64 is supported only for matmul, convolution, reorder, layer normalization, and
```

File: doc/programming_model/data_types.md [hunk 2]
```diff
@@ -43,7 +43,7 @@ oneDNN supports training and inference with the following data types:
...
     pooling primitives, on the GPU engine.
```
Comment (ranukund) [inline doc/programming_model/data_types.md:47]:
```suggestion
    pooling primitives on the GPU engine.
```

**Review by ranukund — APPROVED** [review]
Two minor edits suggested, please incorporate as you see fit. Thanks!

**Review by vpirogov — APPROVED** [review]

**Pull request merged**